### ISSUE TITLE
iot.attachPolicy throws CORS error in browser

### ISSUE THREAD

**Issue by vishalkothari** [aws/aws-sdk-js#3593]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js/issues) and didn't find any solution
  I have looked through #2329 and #1690 and replies from @srchase 

**Describe the bug**
I am trying to dynamically attach IOT policy from javascript using attachPolicy API.
When I use this API, it throws CORS errors.

I am also unable any example or documentation on how to attach cognito identities to IoT policy using attachPolicy API.

**Is the issue in the browser/Node.js?**
Browser

**SDK version number**
Example: v2.824.0

**To Reproduce (observed behavior)**
Steps to reproduce the behavior (please share code or minimal repo)

```
    var policyName = "policy1";
    var iot = new AWS.Iot({
      accessKeyId: AWS.config.credentials.accessKeyId,
      secretAccessKey: AWS.config.credentials.secretAccessKey,
      sessionToken: AWS.config.credentials.sessionToken,
      apiVersion: '2015-05-28'
    });

    var policyParams = {
      "policyName": policyName, /* required */
      "target": AWS.config.credentials.identityId /* required */
    };


    iot.attachPolicy(policyParams, function(err, data) {
      if (err) { console.log(err, err.stack); } // an error occurred
      else {
        console.log("after attach policy");
        console.log(data);  
      }
    });

```

Error
```
Access to XMLHttpRequest at 'https://iot.ap-south-1.amazonaws.com/target-policies/policy1' from origin 'http://localhost:8100' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
PUT https://iot.ap-south-1.amazonaws.com/target-policies/policy1 net::ERR_FAILED
```

I have also tried
```
    var policyParams = {
      "policyName": policyName, /* required */
      "target": "arn:aws:cognito-identity:ap-south-1:<accountid>:identitypool/" + AWS.config.credentials.identityId /* required */
    };
```
But the result is the same.

**Expected behavior**
I have used this before attachPrincipalPolicy and it used work ok for older version of sdk. Now that attachPrincipalPolicy is deprecated, I expect that attachPolicy API will support browsers.

**Screenshots**
NA

**Additional context**
NA

**Comment by ajredniwja** [comment]
Hey @vishalkothari, thanks for opening this issue, I was able to reproduce it, reached out to IOT team to investigate further.

**Comment by trademark18** [comment]
I just ran into this exact same issue.  I'm trying to attach a policy to get access to the IotData api using Cognito authentication.  https://forums.aws.amazon.com/thread.jspa?threadID=247405

**Comment by vishalkothari** [comment]
Hi @ajredniwja, any update from IOT team on this?

**Comment by trademark18** [comment]
Hi @ajredniwja just checking back in on this.  Any word from the IoT team?  This is a blocker for a project at my company, so we are looking forward to hearing what the status is.  Thanks!

**Comment by ajredniwja** [comment]
I have created a new case since I did lose track of the previously created ticket. The workaround maybe to create a proxy to not interact with the service directly.

**Comment by trademark18** [comment]
A workaround that worked for me was to pop open the cloud shell and do the attach policy via the CLI.  It was sufficient to get my website working with the sdk.

```
aws iot attach-policy --policy-name ManagerPolicy --target [identity]
```

Still looking forward to a fix for this.  Thanks!

**Comment by ajredniwja** [comment]
Hey @trademark18 is it still a persisting issue, the team verified that lot supports `attachPolicy`.

**Comment by trademark18** [comment]
@ajredniwja Unfortunately I don't have an easy way to check again.  I've passed off the related project to @ethanx94 who may be able to try it again and report any changes.  I can report that the workaround was a persistent fix, so we just went that route since it's hard to know when the fix for `attachPolicy` may be available.

**Comment by ajredniwja** [comment]
@trademark18, its great that things are working for you, I'd close this issue now, please reach out if anyone faces this problem again or feel free to open a new issue.

**Comment by waleedshkt** [comment]
@ajredniwja, I've checked newer V3 JS SDK docs. In it, it's described about `principal` and `target` input arguments of `attachPrincipalPolicy` and `attachPolicy` APIs respectively.

The former supports cognito identity as input, latter doesn't. 

So are we forced to use certificates?

With multiple cognito identifies, would this mean registering each identity to IOT security certificate, attached to IOT policy and activated, by calling certain API?
